query-planner-js/src/config.ts

```diff
@@ -84,6 +84,16 @@ export type QueryPlannerConfig = {
      */
     maxEvaluatedPlans?: number,
   },
+
+  /**
+   * Before creating query plans, we need to evaluate all the possible paths to access every field:
+   * through which subgraph providing it, through which type in union or type implementing an
+   * interface. The number of paths generated can grow if the schema or query are complex,
+   * and that will increase the time spent planning.
+   *
+   * The default value is set to 10000
+   */
+  pathsLimit: number
```

Comment:
You'll want to use `?` here so that users won't be required to provide the property (and also to not break the API).
```suggestion
  pathsLimit?: number
```
Also could you place this property under the `debug` section for now? (Could potentially move it out later, but for now I'd rather we don't make stability guarantees around it.)